src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorLSPOptionsMonitor.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    internal class RazorLSPOptionsMonitor : IOptionsMonitor<RazorLSPOptions>
```

Comment:
I'm surprised you had to implement a monitor. Isn't a change token source sufficient?

Reply:
I went down that path and realized that I don't just want a trigger to change the option but also to provide the new value myself. It was getting a bit complicated to do that with the change token because I also had to muck with the IOptionsFactory. Implementing IOptionsMonitor turned out to be much simpler plus as a bonus, I can make the options POCO type immutable.